dbschema/partnership.esdl

```diff
@@ -0,0 +1,34 @@
+module default {
+  type Partnership extending Project::Child {
+    agreementStatus: Partnership::AgreementStatus;
+    mou: File; 
+    mouStatus: Partnership::AgreementStatus;
+    mouStart := .mouStartOverride ?? .project.mouStart;
+    mouEnd := .mouEndOverride ?? .project.mouEnd;
+    mouStartOverride: cal::local_date;
+    mouEndOverride: cal::local_date;
+    agreement: File;
+    required partner: Partner;
+    required organization: Organization;
```

Comment:
We don't need to have stored forward links for both of these.
```suggestion
    required partner: Partner;
    organization := .partner.organization;
```
I would ask that these move to the top of the block too since this, `partner`, is one of the keys identifying the `Partnership` (`project` is the other).

Along these lines we need this
```esdl
constraint exclusive on ((.project, .partner));
```
So that a partnership can only be created once for each partner and project.